### ISSUE TITLE
vsplit always opens to the left, can it be on the right? 

### ISSUE THREAD

**Issue by ZacharyRizer** [nvim-tree/nvim-tree.lua#1626]
### Description

When I'm over a file in nvim-tree and press <C-v> to open it in a vsplit, I want the file to open the right of the current file I have open. But it always is placed to the left of the current file. I'm sure this is a config setting I'm missing.

### Neovim version

```text
NVIM v0.9.0-dev-5-g9353e2f7e
Build type: Release
LuaJIT 2.1.0-beta3
Compiled by root@FM-WYN2K2JQ9D
```

### Operating system and version

macOS 12.4

### nvim-tree version

7282f7d

### Minimal config

```lua
vim.opt.splitbelow = true
vim.opt.splitright = true

local tree_cb = require 'nvim-tree.config'.nvim_tree_callback
require('nvim-tree').setup({
    actions             = {
        open_file = { quit_on_open = true },
    },
    renderer            = {
        indent_markers = { enable = true }
    },
    respect_buf_cwd     = true,
    update_cwd          = true,
    update_focused_file = { enable = true, update_cwd = true },
    view                = {
        adaptive_size = true,
        mappings = {
            custom_only = false,
            list = {
                { key = "<CR>", cb = tree_cb("edit") },
                { key = "h", cb = tree_cb("close_node") },
                { key = "l", cb = tree_cb("open_node") },
                { key = "<BS>", cb = tree_cb("dir_up") },
                { key = "?", cb = tree_cb("toggle_help") },
                { key = "<C-e>", cb = tree_cb("close") },
                { key = "<C-s>", cb = tree_cb("split") },
            }
        },
        width = 35,
        side = 'right',
    },
})
```

### Steps to reproduce

1. have a file open in vim buffer `current_file`
2.  `:NvimTreeOpen`
3. navigate to file
4. `<C-v>` to open in vsplit

### Expected behavior

File opens to the right of `current_file`

### Actual behavior

File opens to the left of `current_file`

**Comment by kyazdani42** [comment]
see `:help splitright`

**Comment by ZacharyRizer** [comment]
So as I showed in the original post in the minimal init.lua, I have splitbelow and splitright both set to true. But the buffer still opens to the left. @kyazdani42

**Comment by alex-courtis** [comment]
Boiled this one down:

Default setup with `vim.opt.splitright = false` and `vim.opt.splitbelow = false`

`<C-x>` opens the new window below.
`<C-v>` opens the new window to the right.

This is a regression.

**Comment by kyazdani42** [comment]
I use nvim-tree on the right, i do set splitright to false, it work just fine though.

**Comment by kyazdani42** [comment]
well so this is expected, since when opening, depending on where the tree should be placed, the `vsplit` action will split to let the tree stay where it supposed to be.
```lua
  local do_split = mode == "split" or mode == "vsplit"
  local split_side = (view.View.side == "right") and "aboveleft" or "belowright"
```
This is expected behavior, since why would you open file on the left if you want to set the tree to be on the right ?
Closing, if you need to introduce a new behavior, please open a new PR :)

**Comment by kassio** [comment]
> This is expected behavior, since why would you open file on the left if you want to set the tree to be on the right ?

I have the tree in the right and also have `splitright` set. I would expect to hit `c-v` and open the split on the right of the last focused file (not on the right of the tree). 

How do I achieve this behaviour without changing the `splitright`, since I also split manually or with other plugins that I'd like to keep respecting this value?

**Comment by alex-courtis** [comment]
> I have the tree in the right and also have `splitright` set. I would expect to hit `c-v` and open the split on the right of the last focused file (not on the right of the tree).

Also a picked window.

**Comment by alex-courtis** [comment]
I'm not quite sure things are working correctly with `splitright`, regardless of view.side. The exectation here is that the split occurs relative to the last or picked window rather than the nvim-tree window.

Here's what normally happens when using `<C-w>v`
`splitright` is obeyed relative to the current window.
[![asciicast](https://asciinema.org/a/527227.svg)](https://asciinema.org/a/527227)

Here's a similar case with nvim-tree defaults `<C-v>`
`splitright` is not set, however the new window is always opened on the right.
[![asciicast](https://asciinema.org/a/527228.svg)](https://asciinema.org/a/527228)

**Comment by alex-courtis** [comment]
5cb87c0 is the cause of the regression.

**Comment by alex-courtis** [comment]
fix branch: [1626-open-file-split-not-obeying-splitright-splitbelow](https://github.com/nvim-tree/nvim-tree.lua/tree/1626-open-file-split-not-obeying-splitright-splitbelow)
PR: #1641

@ZacharyRizer I would be very grateful if this fix were tested:
```
cd /path/to/nvim-tree.lua
git pull
git checkout 1626-open-file-split-not-obeying-splitright-splitbelow
```
Checkout master or nightly when you are finished testing.

**Comment by ZacharyRizer** [comment]
Just pulled that branch and tested. Seems to be working for me! @alex-courtis

**Comment by naquad** [comment]
Can we have a setting controlling split_side instead of deriving it from the view side?

**Comment by alex-courtis** [comment]
> Can we have a setting controlling split_side instead of deriving it from the view side?

It is possible, however the use case is not obvious. What is the motivation for this?

**Comment by naquad** [comment]
> It is possible, however the use case is not obvious. What is the motivation for this?

The current solution tries to open splits on the opposite side of the view (view on the left side - open a split on the right below, right - open on the top left). It is inconvenient as when there's a big screen you need to look at another corner of the screen. Also, as a person coming from NERDTRee it feels very awkward for me.

**Comment by alex-courtis** [comment]
Unfortunately I don't think I understand correctly. Is this the proposal?

side left with one buffer open
```
-----------------------------------------------
|            |                                |
| nvim tree  | buffer 1                       |
|            |                                |
|            |                                |
|            |                                |
|            |                                |
|            |                                |
|            |                                |
|            |                                |
|            |                                |
|            |                                |
|            |                                |
|            |                                |
-----------------------------------------------
```

current behaviour on `<C-x>`
```
-----------------------------------------------
|            |                                |
| nvim tree  | buffer 1                       |
|            |                                |
|            |                                |
|            |                                |
|            |--------------------------------|
|            |                                |
|            |                                |
|            | buffer 2                       |
|            |                                |
|            |                                |
|            |                                |
|            |                                |
-----------------------------------------------
```

proposed behaviour on `<C-x>` with split_side left
```
-----------------------------------------------
|            |            |                   |
| buffer 2   | nvim tree  | buffer 1          |
|            |            |                   |
|            |            |                   |
|            |            |                   |
|            |            |                   |
|            |            |                   |
|            |            |                   |
|            |            |                   |
|            |            |                   |
|            |            |                   |
|            |            |                   |
|            |            |                   |
-----------------------------------------------
```

**Comment by naquad** [comment]
No, that's not the case, but things got extremely mixed here. I'm not sure if that's the bug but here the asciicast of what's happening:

[![asciicast](https://asciinema.org/a/rV8QrYRfYGDZ2L0IWTtVMcdvo.svg)](https://asciinema.org/a/rV8QrYRfYGDZ2L0IWTtVMcdvo)

My setup configuration:

```

nvim_tree.setup {
  git = {
    ignore = false
  },
  filters = {
    dotfiles = true,
  },
  view = {
    side = 'right',
    mappings = {
      list = {
        { key = '<CR>',  action = 'edit_no_picker',     },
        { key = '<Tab>', action = 'split',              },
        { key = 'I',     action = 'toggle_dotfiles',   },
        { key = 'H',     action = 'toggle_git_ignored', },
        {
          key = '<Leader>gg',
          mode = 'n',
          action = 'print_path',
          action_cb = grepper_here,
        }
      }
    }
  },
  actions = {
    open_file = {
      window_picker = {
        enable = false
      }
    }
  }
}
```

**Comment by alex-courtis** [comment]
The above doesn't occur after #1641

Are you running the latest?

**Comment by naquad** [comment]
I've been running nightly, sorry. I've switched to master and pulled. Things got... awkward:

[![asciicast](https://asciinema.org/a/5ixluMLp0UFDgbNnoAXfRanC2.svg)](https://asciinema.org/a/5ixluMLp0UFDgbNnoAXfRanC2)

**Comment by alex-courtis** [comment]
> I've been running nightly, sorry. I've switched to master and pulled. Things got... awkward

Wow... that went badly.

Please raise a new [bug report](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=bug&template=bug_report.yml) with an nvt-min.lua and steps so that I may reproduce.

A "clean room" in which to replicate is necessary which removes other plugins and automation, having only the minimal necessary configuration changes.
